atlasdb-workload-server/src/test/java/com/palantir/atlasdb/workload/store/AtlasDbTransactionStoreFactoryTest.java

```diff
@@ -0,0 +1,152 @@
+    }
+
+    @Test
+    public void toAtlasTablesConvertsSerializableTablesCorrectly() {
+        assertThat(defaultFactory.toAtlasTables(
+                        Map.of(WorkloadTestHelpers.TABLE, IsolationLevel.SERIALIZABLE), Set.of()))
+                .containsExactlyInAnyOrderEntriesOf(Map.of(
+                        defaultFactory.createTableReference(WorkloadTestHelpers.TABLE),
+                        AtlasDbUtils.tableMetadata(ConflictHandler.SERIALIZABLE)));
+    }
+
+    @Test
+    public void toAtlasTablesConvertsSerializableIndexTablesCorrectly() {
+        assertThat(defaultFactory.toAtlasTables(
+                        Map.of(WorkloadTestHelpers.TABLE, IsolationLevel.SERIALIZABLE),
+                        Set.of(IndexTable.of(WorkloadTestHelpers.INDEX_TABLE, WorkloadTestHelpers.TABLE))))
+                .containsExactlyInAnyOrderEntriesOf(Map.of(
+                        defaultFactory.createTableReference(WorkloadTestHelpers.TABLE),
+                        AtlasDbUtils.tableMetadata(ConflictHandler.SERIALIZABLE),
+                        defaultFactory.createTableReference(WorkloadTestHelpers.INDEX_TABLE),
+                        AtlasDbUtils.indexMetadata(ConflictHandler.SERIALIZABLE)));
+    }
+
+    @Test
+    public void toAtlasTablesHandlesMultipleTablesAndIndexesCorrectly() {
+        assertThat(defaultFactory.toAtlasTables(
+                        Map.of(
+                                WorkloadTestHelpers.TABLE,
+                                IsolationLevel.SERIALIZABLE,
+                                WorkloadTestHelpers.TABLE_2,
+                                IsolationLevel.SNAPSHOT),
+                        Set.of(
+                                IndexTable.of(WorkloadTestHelpers.INDEX_TABLE, WorkloadTestHelpers.TABLE),
+                                IndexTable.of(WorkloadTestHelpers.INDEX_2_TABLE, WorkloadTestHelpers.TABLE),
+                                IndexTable.of(WorkloadTestHelpers.INDEX_TABLE_2, WorkloadTestHelpers.TABLE_2))))
+                .containsExactlyInAnyOrderEntriesOf(Map.of(
+                        defaultFactory.createTableReference(WorkloadTestHelpers.TABLE),
+                        AtlasDbUtils.tableMetadata(ConflictHandler.SERIALIZABLE),
+                        defaultFactory.createTableReference(WorkloadTestHelpers.INDEX_TABLE),
+                        AtlasDbUtils.indexMetadata(ConflictHandler.SERIALIZABLE),
+                        defaultFactory.createTableReference(WorkloadTestHelpers.INDEX_2_TABLE),
+                        AtlasDbUtils.indexMetadata(ConflictHandler.SERIALIZABLE),
+                        defaultFactory.createTableReference(WorkloadTestHelpers.TABLE_2),
+                        AtlasDbUtils.tableMetadata(ConflictHandler.RETRY_ON_WRITE_WRITE),
+                        defaultFactory.createTableReference(WorkloadTestHelpers.INDEX_TABLE_2),
+                        AtlasDbUtils.tableMetadata(ConflictHandler.IGNORE_ALL)));
```

Comment:
maybe here, though preferably in `AtlasDbUtilsTest`: the mapping of `IsolationLevel.NONE` to `IGNORE_ALL` should be tested